emiliopavia/io-app
# Add the wizard for requesting the account deletion

### ⚠️ This pr depends on https://github.com/emiliopavia/io-app/pull/5 ⚠️
#### Please review https://github.com/emiliopavia/io-app/pull/5 first in order to avoid duplicate review work.

## Short description
When the user enables the account deletion switch, a wizard will guide the user in the process. The same will happen from a button available in the wallet main screen.

## List of changes proposed in this pull request
- `UserProfileScreen` has been refactored to extract a `UserProfileDetails` component that will be reused in another screen
- Added 3 new screens that implement the wizard: `ProfileDeletionStartScreen`, `ProfileDeletionConfirmScreen` and `ProfileDeletionSuccessScreen`
- Added `ProfileDeletionNavigator` to navigate throughout the above screens
- The above navigator has been added to both `ProfileNavigator` and `WalletNavigator`
- `UserDeleteWalletButton` has been implemented and added to `WalletHomeScreen`
- `profileDeletionWorkUnit` has been defined and launched from the `watchUserProfileSaga`

## How to test
Go to the profile info screen and enable the switch to start the deletion wizard. Also test the process starting from the button in the wallet screen. At the end of the process the user should go back where it started.

| Wallet CTA |
| - |
| <img src="https://user-images.githubusercontent.com/467098/156724539-472df01c-a18a-4867-81dd-3cefe54fa0fc.png" width="280"/> |

https://user-images.githubusercontent.com/467098/156724587-7a600878-0de9-4930-b8d5-c924bd46c4a0.mov
